model/modelprocessor/generate_internal_metrics.go

```diff
@@ -0,0 +1,99 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build tools
+// +build tools
+
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"text/template"
+
+	"gopkg.in/yaml.v3"
+)
+
+func main() {
+	_, file, _, ok := runtime.Caller(0)
+	if !ok {
+		log.Fatal("runtime.Caller failed")
+	}
+	modelprocessorDir := filepath.Dir(file)
+	repoRoot := filepath.Join(modelprocessorDir, "..", "..")
+
+	internalMetricsFieldsDir := filepath.Join(repoRoot, "apmpackage", "apm", "data_stream", "internal_metrics", "fields")
+	files, err := filepath.Glob(filepath.Join(internalMetricsFieldsDir, "*_metrics.yml"))
```

Comment:
Are we going to keep the `yaml` definitions of the data stream fields in sync with what the agents send in a best-effort manual fashion?

Reply:
Yes. The metrics change pretty infrequently. Later on we might revert to dynamically mapping metrics identified as being internal, only in the internal metrics data stream.